int64 start_id
int64 end_id
int64 observe_x
int64 observe_y
